src/main/java/org/mskcc/oncokb/curation/service/CategoricalAlterationService.java

```diff
@@ -62,6 +66,13 @@ public Optional<CategoricalAlteration> partialUpdate(CategoricalAlteration categ
             .map(categoricalAlterationRepository::save);
     }
 
+    public Optional<CategoricalAlteration> findByAlterationTypeAndName(AlterationType alterationType, String name) {
+        if (alterationType == null) {
+            return Optional.empty();
+        }
+        return categoricalAlterationRepository.findByAlterationTypeAndName(alterationType, name);
```

Comment:
Shouldn't the spring jpa method already return empty optional if type is null?

Reply:
I want to avoid sending call when alteration type is null